tests/dns.rs

```diff
@@ -0,0 +1,110 @@
+use std::{
+    io::{BufRead, BufReader, Write},
+    net::TcpStream,
+    process::{Child, Command},
+    sync::{
+        mpsc,
+        mpsc::{Receiver, Sender},
+    },
+    thread,
+    time::Duration,
+};
+
+fn start_server() -> (Child, Receiver<String>) {
+    let (log_sender, log_receiver): (Sender<String>, Receiver<String>) = mpsc::channel();
+    let mut directoryd_process = Command::new("./target/debug/directoryd")
+        .stdout(std::process::Stdio::piped())
+        .spawn()
+        .unwrap();
+
```

Comment:
I dont have any opinions on this, can be added, or we can just add a comment, starting the daemon with default values both works..

Reply:
this can't be implemented. once we shutdown and restart the server the temp dir is lost and a new one is created, so we can't do persistence check in the test.